rockcraft/project.py

```diff
@@ -96,15 +110,15 @@ class Platform(pydantic.BaseModel):
 
     @pydantic.validator("build_for", pre=True)
     @classmethod
-    def _vectorise_build_for(cls, val):
+    def _vectorise_build_for(cls, val: Any) -> Any:
```

Comment:
Are we expecting to handle anything less strict than:

```suggestion
    def _vectorise_build_for(cls, val: Union[str, List[str]]) -> List[str]:
```

Reply:
I was undecided on doing that because this (specific) validator runs before the typical Pydantic validation of the field types, so `val` here is going to be whatever the used put in the `build_for` field (also I guess I got a bit lazy)
But then that's the case for every type annotation for function parameters in Python, so I changed it as you suggested, thanks!